lib/srv/db/dynamodb/engine.go

```diff
@@ -329,55 +333,128 @@ func (e *Engine) getRoundTripper(ctx context.Context, URL string) (http.RoundTri
+// Targets look like one of DynamoDB_$version.$operation,
+// DynamoDBStreams_$version.$operation, or AmazonDAX$version.$operation.
+// For example: DynamoDBStreams_20120810.ListStreams
+func (e *Engine) resolveEndpoint(req *http.Request) (*endpoint, error) {
+	target, err := getTargetHeader(req)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
-	opts := func(opts *endpoints.Options) {
-		opts.ResolveUnknownService = true
+
+	awsMeta := e.sessionCtx.Database.GetAWS()
+
+	var re *endpoint
+	switch target := strings.ToLower(target); {
+	case strings.HasPrefix(target, "dynamodbstreams"):
+		re, err = resolveDynamoDBStreamsEndpoint(req.Context(), awsMeta.Region, e.UseFIPS)
+	case strings.HasPrefix(target, "dynamodb"):
+		re, err = resolveDynamoDBEndpoint(req.Context(), awsMeta.Region, awsMeta.AccountID, e.UseFIPS)
+	case strings.HasPrefix(target, "amazondax"):
+		re, err = resolveDaxEndpoint(req.Context(), awsMeta.Region, e.UseFIPS)
+	default:
+		return nil, trace.BadParameter("DynamoDB API target %q is not recognized", target)
 	}
-	re, err := endpoints.DefaultResolver().EndpointFor(endpointID, e.sessionCtx.Database.GetAWS().Region, opts)
 	if err != nil {
 		return nil, trace.Wrap(err)
 	}
 	uri := e.sessionCtx.Database.GetURI()
-	if uri != "" && uri != apiaws.DynamoDBURIForRegion(e.sessionCtx.Database.GetAWS().Region) {
+	if uri != "" && uri != apiaws.DynamoDBURIForRegion(awsMeta.Region) {
+		// Add a temporary schema to make a valid URL for url.Parse.
+		if !strings.Contains(uri, "://") {
+			uri = "schema://" + uri
+		}
+		u, err := url.Parse(uri)
+		if err != nil {
+			return nil, trace.Wrap(err)
+		}
 		// override the resolved endpoint URL with the user-configured URI.
-		re.URL = uri
+		re.URL = u
 	}
-	if !strings.Contains(re.URL, "://") {
-		re.URL = "https://" + re.URL
+	// Force HTTPS
+	re.URL.Scheme = "https"
+	return re, nil
+}
+
+func resolveDynamoDBStreamsEndpoint(ctx context.Context, region string, useFIPS bool) (*endpoint, error) {
+	params := dynamodbstreams.EndpointParameters{
+		Region:  aws.String(region),
+		UseFIPS: aws.Bool(useFIPS),
+	}
+	ep, err := dynamodbstreams.NewDefaultEndpointResolverV2().ResolveEndpoint(ctx, params)
+	if err != nil {
+		return nil, trace.Wrap(err)
 	}
-	return &re, nil
+	return &endpoint{
+		URL:           &ep.URI,
+		SigningRegion: region,
+		// DynamoDB Streams uses the same signing name as DynamoDB.
+		SigningName: "dynamodb",
+	}, nil
 }
 
-// rewriteRequest clones a request, modifies the clone to rewrite its URL, and returns the modified request clone.
-func rewriteRequest(ctx context.Context, r *http.Request, re *endpoints.ResolvedEndpoint, body []byte) (*http.Request, error) {
-	resolvedURL, err := url.Parse(re.URL)
+func resolveDynamoDBEndpoint(ctx context.Context, region, accountID string, useFIPS bool) (*endpoint, error) {
+	params := dynamodb.EndpointParameters{
+		Region:                aws.String(region),
+		AccountIdEndpointMode: aws.String(aws.AccountIDEndpointModePreferred),
```

Comment:
what does this mode do? assuming it's giving `https://123456789012.ddb.us-west-1.amazonaws.com`. is the account id endpoint always available? any reference?